<launch>

  <!-- ========== MOCAP_OPTITRACK ========== -->
  <arg name="mocap_config_file"
    default="$(find mocap_interface)/config/mocap.yaml" />
  <node pkg="mocap_optitrack" type="mocap_node" name="mocap_node" respawn="false" launch-prefix="" required="true" output="screen">
    <rosparam file="$(arg mocap_config_file)" command="load" />
  </node>

	<!-- ========== JOY ========== -->
  <node name="joy_node" pkg="joy" type="joy_node" output="screen">
  </node>
  <!-- ========== OUIJABOT TELEOP ========== -->
  <node name="telop" pkg="ouijabot_telop_cpp" type="telop"  output="screen">
    <remap from="vel_out" to="/ouijabot1/cmd_vel" /> <!-- change the index of the robot here -->
  </node>

  <!-- ========== OUIJABOT ODOMETRY ========== -->
  <node name="filter" pkg="ouijabot_filter" type="filter"  output="screen">
    <remap from="/robot/pose" to="/robot_1/pose"/>
    <remap from="/ouijabot/imu" to="/ouijabot1/imu"/>
    <remap from="/ouijabot/wheel_spd" to="/ouijabot1/wheel_spd"/>
    <remap from="/robot/state_estimate" to="/robot_1/state_estimate"/>
    <param name="DISPLAY_DATA_BOOL" type="int" value="1" />
    <param name="WHEEL_RADIUS" type="double" value="0.025" />
    <param name="BODY_RADIUS" type="double" value="0.1" />
    <param name="POS_COVARIANCE" type="double" value="0.1" />
    <param name="ORIENT_COVARIANCE" type="double" value="0.1" />
    <param name="VEL_COVARIANCE" type="double" value="0.001" />
    <param name="ROT_RATE_COVARIANCE" type="double" value="0.001" />
    <param name="ACCEL_COVARIANCE" type="double" value="0.0001" />
    <param name="ROT_ACCEL_COVARIANCE" type="double" value="0.0001" />
    <param name="WHEEL_MEASUREMENT_COVARIANCE" type="double" value="0.1" />
    <param name="GYRO_MEASUREMENT_COVARIANCE" type="double" value="0.01" />
    <param name="ACCEL_MEASUREMENT_COVARIANCE" type="double" value="0.01" />
  </node>
  <!-- ========== LINK BETWEEN ODOMETRY GROUND TRUTH AND WORLD WORLD ========== -->
  <node pkg="tf" type="static_transform_publisher" name="trans_pub_world_odom_gt"
    args="0 0 0 0 0 0 /world /odom_gt_frame 100">
  </node>
  <!-- ========== LINK BETWEEN ODOMETRY ESTIMATOIN AND WORLD WORLD ========== -->
  <node pkg="tf" type="static_transform_publisher" name="trans_pub_world_odom"
    args="0 0 0 0 0 0 /world /odom_frame 100">
  </node>

  <!-- ========== RVIZ ========== -->
  <node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen" 
    args="-f world_rviz -d $(find ouijabot_odom_cpp)/rviz/odom_testing.rviz" required="true">
  </node>
  <!-- ========== LINK BETWEEN MOCAP_OPTITRACK WORLD AND RVIZ WORLD ========== -->
  <node pkg="tf" type="static_transform_publisher" name="trans_pub_rviz_world"
    args="0 0 0 1.5708 0 3.1415 /world_rviz /world 100">
  </node>

</launch>
